src/libraries/Microsoft.PowerFx.Core/Public/Values/UntypedObjectValue.cs

```diff
@@ -28,7 +28,9 @@ public interface IUntypedObject
         /// </summary>
         IUntypedObject this[int index] { get; }
 
-        bool TryGetProperty(string value, out IUntypedObject result);
+        bool TryGetProperty(string value, out IUntypedObject result);
+
+        bool TrySetProperty(string name, IUntypedObject value);
```

Comment:
>TrySetProperty [](http://example.com/codeflow?start=13&length=14)

Mecahnics issue: technically adding to interface is a breaking change. I'm wondering if we should add these as virtuals UntypedObjectValue instead. That's a class, so it's non-breaking. We can discuss offline